application/src/main/scala/com/azavea/franklin/tile/CogAssetNodeImplicits.scala

```diff
@@ -12,47 +13,39 @@ import geotrellis.vector.Extent
 
 object CogAssetNodeImplicits extends TileUtil {
 
-  implicit def cogAssetNodeTmsReification: TmsReification[CogAssetNode] =
-    new TmsReification[CogAssetNode] {
-
-      def tmsReification(
-          self: CogAssetNode,
-          buffer: Int
-      )(implicit cs: ContextShift[IO]): (Int, Int, Int) => IO[ProjectedRaster[MultibandTile]] =
-        (z: Int, x: Int, y: Int) => {
-          def fetch(xCoord: Int, yCoord: Int): IO[Raster[MultibandTile]] = {
-            self.fetchTile(z, xCoord, yCoord, WebMercator).flatMap(a => IO(a))
-          }
-
-          fetch(x, y).map { tile =>
-            val extent = tmsLevels(z).mapTransform.keyToExtent(x, y)
-            ProjectedRaster(tile.tile, extent, WebMercator)
-          }
+  implicit def cogAssetNodeTmsReification[F[_]: Sync]: TmsReification[F, CogAssetNode] = {
+    (self, _) => (z: Int, x: Int, y: Int) =>
+      {
+        def fetch(xCoord: Int, yCoord: Int): F[Raster[MultibandTile]] = {
+          self.fetchTile(z, xCoord, yCoord, WebMercator).flatMap(a => Sync[F].delay(a))
```

Comment:
I think the flatmap is unnecessary, since it's basically `x flatMap { a => pure(a) }`, so it just gives the scheduler I tiny bit more work to do. It's a faithful translation but a really interesting original choice 🤔

```suggestion
          self.fetchTile(z, xCoord, yCoord, WebMercator)
```

Reply:
@jisantuc lool, thanks for your review, I missed that!